### ISSUE TITLE
add_to with name is not passing the name in to Folium

### ISSUE THREAD

**Issue by alexgleith** [opendatacube/odc-geo#108]
I'm adding a couple of xarray variables to a folium map, and it works fantastically but I can't get it to name them neatly.

Here's my code:

```python
import folium
import odc.geo.xr

m = folium.Map(control_scale=True, tiles=None)

data.vv.odc.add_to(m, opacity=0.7, name="vv")
data.vh.odc.add_to(m, opacity=0.7, name="vh")

# Zoom map
m.fit_bounds(data.odc.map_bounds())

tile = folium.TileLayer(
    tiles = 'https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}',
    attr = 'Esri',
    name = 'Esri Satellite',
    overlay = False,
    control = True
).add_to(m)
folium.TileLayer('openstreetmap').add_to(m)

folium.LayerControl().add_to(m)
display(m)
```

I have the following software versions:

* `folium==0.14.0`
* `odc-geo==0.4.0`

And here's how the layers display:

![image](https://github.com/opendatacube/odc-geo/assets/3445853/b69d7b41-b2e4-4306-89e1-c29aeabb6ee7)

**Comment by Kirill888** [comment]
@alexgleith this is a dupe of #102 and has been fixed in #103, we need to make a release with that fix.

**Comment by Kirill888** [comment]
can you please try with code in `develop` branch and see if that issue is fixed for your example?

**Comment by alexgleith** [comment]
Ok, tested.

Installed from git `pip3 install --upgrade git+https://github.com/opendatacube/odc-geo.git@develop`

Results in:
`pip freeze | grep odc-geo`
                                            
> odc-geo @ git+https://github.com/opendatacube/odc-geo.git@53535c8dfd58dd10e7e93c676055a57fb3ec95cd

Now restarting a notebook kernel and re-running gives me a nicely labelled layer.

Thanks Kirill!

![image](https://github.com/opendatacube/odc-geo/assets/3445853/03adc6e3-d1a7-4c04-8fcd-a2024078168a)

**Comment by Kirill888** [comment]
@alexgleith thanks for testing, this is now released as `odc-geo==0.4.1` on both pypi and conda-forge
